src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -395,16 +395,21 @@ getWAddressBalance addr =
     getBalance <=< decodeCIdOrFail $ cwamId addr
 
 getWAddress :: WalletWebMode m => CWAddressMeta -> m CAddress
-getWAddress cAddr@CWAddressMeta {..} = do
+getWAddress cAddr = do
+    let aId = cwamId cAddr
     balance <- getWAddressBalance cAddr
+    let addrAccount = addrMetaToAccount cAddr
     (ctxs, _) <-
         searchHistory
             Nothing
-            (Just $ addrMetaToAccount cAddr)  -- just to specify addrId is not enough
-            (Just cwamId)
+            (Just addrAccount)  -- just to specify addrId is not enough
+            (Just aId)
             Nothing
     let isUsed = not (null ctxs) || balance > minBound
-    CAddress cwamId (mkCCoin balance) isUsed <$> isChangeAddress cAddr
+    -- Suppose we have transaction with inputs A1, A2, A3. Output address B_j is referred to as change address if it belongs to same account as one of A_i
+    acctAddrs <- map cwamId <$> getAccountAddrsOrThrow Ever addrAccount
+    let isChange = elem aId $ filter (`elem` acctAddrs) $ concatMap ctOutputAddrs ctxs
```

Comment:
1. `filter (`elem` acctAddrs)` seems redundant because it's implied by `elem aId` - if `x == aId`, then `x` indeed belongs to the set of all ever existed accounts `acctAddrs`.
2. This seem not to work with example above. For instance, if `cAddr` is `B_j` from example, then `aId` indeed belongs to `concatMap ctOutputAddrs ctxs` which is set of outputs of all related transactions, but we want `cAddr` to be considered _change_ only if it participates as input of some transaction as well